### DISCUSSION TITLE
How to create msixbundle packages for Store

### DISCUSSION THREAD

**Discussion by JanRajnoha** [microsoft/WindowsAppSDK#2811] [Q&A]
I have Win App SDK app (1.1.3), that I want to package, but when I use "Package and Publish" option in menu, go through wizard without problem, wait for result, which is without errors too, I have only MSIX packages, that are certified with temporary certificate, not the Store one.

Is it bug in Windows App SDK?

**Comment by JanRajnoha** [comment]
@gabbybilka Could you please help?

**Reply by gabbybilka** [reply]
@DrusTheAxe could you take a look? Thanks!

**Reply by JanRajnoha** [reply]
@DrusTheAxe Could you please take a look? Thank you very much :)

**Comment by EFLKumo** [comment]
But why do u need `.msixbundle` file(s)? If you want to deploy your application on computers, right click your project or packaging project, click `Package` then goto `bin\{You CPU Arch, may be x64 or x86 or arm64}\{Debug or Release}\net6.0-windows10.0.xxxxx\win10-xxx\AppPackages\`, you will see a folder, get in, and get sth like:  

![image](https://user-images.githubusercontent.com/96906415/183042971-a0d39c1d-3734-42bc-98a0-c69e36ac9afa.png)  
Just run `Install.ps1` on your computer and it will deploy your applicaton.  
*****
If you want to deploy your application on users' PCs, not developers' PCs, you should publish application on Microsoft Store.

**Reply by JanRajnoha** [reply]
@RayGicEFL yeah, I could install it via ps1, but i want to distribute it with Store. Problem with msix is certificate, which is temporary created in Visual Studio, so I couldn't upload it to Store. Msix bundle however everytime create bundle, which has Store acceptable certificate

**Reply by EFLKumo** [reply]
Err. You're right. But although with msix bundle, you still must have a no-temporary certificate and for most of time, it costs money. That means you have to **buy** a certificate and change your package certificate to that. So then user can install your apps via ms store. Actually, I don't understand why "Msix bundle however everytime create bundle, which has Store acceptable certificate" because I believe microsoft is not so conscience that will provide free one for you!

**Reply by JanRajnoha** [reply]
I have Store account, I already publish app, but with old app-package model. When you have store account, you could associate your app with store account and everything is created okay ready to publish to Store. With newer version of WASDK I could use single app model, so I try create package, that was okay, but msixbundle missed. So certificate you "get" from store account (probably, i don't know, how it works, but I did it everytime same since Win8 - Associate with store), but now it is problem.

**Reply by EFLKumo** [reply]
OK..I've never published my app on Store before so I don't know how it works, and it seems that this issue need someone else to help..

**Reply by JanRajnoha** [reply]
oh, if you have something, you have to publish it. Store needs more and more great apps ... just you have to pay one time fee ($19)

**Reply by EFLKumo** [reply]
> oh, if you have something, you have to publish it. Store needs more and more great apps ... just you have to pay one time fee ($19)

and I just don't have the one time fee... do u think that a junior high school student's parents will agree that?

**Reply by JanRajnoha** [reply]
If you love coding, why won't they agree and for china is fee 116 CNY.

**Reply by DrusTheAxe** [reply]
+ @MikeHillberg

**Comment by EFLKumo** [comment]
Do you tried to adjust `<AppxBundle>` property under `<PropertyGroup>` in your project file?

**Reply by EFLKumo** [reply]
![image](https://user-images.githubusercontent.com/96906415/184471736-61a7a14b-1bee-407c-a2d4-f786573001f3.png)
It's `Never` in mine so that might be the reason.

**Reply by JanRajnoha** [reply]
I have it always,  so this one could not be problem :/

**Comment by rocksdanister** [comment]
I don't think WinAppSDK supports mxibundle building yet(?), this is what I use to build msixbundle from msix using cmd:
` "C:\Program Files (x86)\Windows Kits\10\bin\10.0.18362.0\x86\MakeAppx.exe" bundle /d "C:\Source" /p C:\Bundle\out.msixbundle`

Put all the msix files in the `Source` folder.

**Reply by EFLKumo** [reply]
Thanks!!

**Reply by JanRajnoha** [reply]
I'll try and will inform you about it

**Reply by JanRajnoha** [reply]
Your command is okay, it will create bundle, but there is still problem with temporary certificate used for creating msix packages, so I can't still upload app to Store :(

**Reply by rocksdanister** [reply]
I don't understand? store will sign the bundle once uploaded.

**Reply by JanRajnoha** [reply]
No, when I want to upload package, I get error message about wrong publisher and package name, even if i have it set correctly in Visual Studio. Values, that are taken for publisher and package name are from temporary certificate. I tried to remove it from project and result was same. I found the problem and solution for this problem.

**Comment by JanRajnoha** [comment]
Workaround
---
I finally found workaround for this problem. These steps are same result as process in Visual Studio, but it is not solution for problem. So, where were problems?

**Certificate**
Even after association app with Store, package process use old temporary certificate. After removing certificate form project, folder, everything was this certificate used. In appxmanifest is link directly to temporary certificate to certification store. Solution I found in certification store, where I have Store certificate used for production app, so currently I could publish it to Store.

**MSIX bundle**
Here are creadits going to @rocksdanister and his command., thank you.

But still, I will be happy for response from Microsoft. @MikeHillberg

**Reply by myokeeh** [reply]
I'm having the same problem. I do not follow what needs to be done to the certificate. Remove it completely and create a new one?

**Reply by rocksdanister** [reply]
If you have temporary certificate remove it from the .csproj file.

**Reply by myokeeh** [reply]
Ok I think I got it. Also, when running MakeAppx.exe, it appears it's important that the "Source" folder only has msix files.
